### ISSUE TITLE
--manifest-path foo/fake works

### ISSUE THREAD

**Issue by huonw** [rust-lang/cargo#1773]
```
$ cargo new test
$ cargo build --manifest-path test/fake
   Compiling test v0.1.0 (file:///home/huon/projects/test-rust/test)
```

This seems fairly peculiar. It looks a bit like it strips the last segment of the path without looking at it at all.

**Comment by cybergeek94** [comment]
Sounds like it assumes the last path segment is `Cargo.toml` and pops it unconditonally, then it appends `Cargo.toml` back to the new path. It might as well just take the directory to search in if it's going to ignore the actual manifest path. Do we want to support manifest files named something other than `Cargo.toml`? If not, this should be changed to `--manifest-dir` and the value should be read as a directory.

**Comment by alexcrichton** [comment]
This option has always confused me and it's been an endless source of confusion in terms of good error messages and such. I'm honestly not 100% sure what the original intention for the flag was, @wycats do you remember what the precise semantics of `--manifest-path` were supposed to be?

**Comment by carols10cents** [comment]
Soooo funny story... there's [this comment](https://github.com/rust-lang/cargo/issues/1158#issuecomment-85267210) on a similar issue that says "I believe this was fixed in #1182.", but #1182 is a pull request that was closed without merging....

I've just opened #1955 that is heavily influenced by #1158 and should also fix this issue :)

**Comment by alexcrichton** [comment]
Holy cow, that would explain why I'm always so confused about why these issues come up, I could have sworn I fixed them! I'm... not sure what happened to #1182? Maybe I accidentally deleted the branch? Regardless #1955 is looking good to me!

**Comment by carols10cents** [comment]
Since #1955 got merged, I think this is fixed:

```
$ cargo new test
$ cargo build --manifest-path test/fake
the manifest-path must be a path to a Cargo.toml file
```

Sorry, I should have put a "Fixes #1773" in that issue :)

**Comment by alexcrichton** [comment]
Thanks @carols10cents!
